File: test/benchmark/python/zoo.py
import time


class Zoo:
    def __init__(self):
        self.aarvark = 1
        self.baboon = 1
        self.cat = 1
        self.donkey = 1
        self.elephant = 1
        self.fox = 1

    def ant(self):
        return self.aarvark

    def banana(self):
        return self.baboon

    def tuna(self):
        return self.cat

    def hay(self):
        return self.donkey

    def grass(self):
        return self.elephant

    def mouse(self):
        return self.fox


zoo = Zoo()
s = 0
start = time.process_time()

while s < 10000000:
    s = (
        s
        + zoo.ant()
        + zoo.banana()
        + zoo.tuna()
        + zoo.hay()
        + zoo.grass()
        + zoo.mouse()
    )

print(s)
print(time.process_time() - start)

File: test/benchmark/ruby/zoo.rb
class Zoo
    #Python fields are always public whereas Ruby's are private. attr_accessor makes them public
    attr_accessor :cat, :donkey, :elephant, :aarvark, :baboon, :fox
    def initialize()
        @aarvark = 1
        @baboon = 1
        @cat = 1
        @donkey = 1
        @elephant = 1
        @fox = 1
    end

    def ant()
        return @aarvark
    end

    def banana()
        return @baboon
    end

    def tuna()
        return @cat
    end

    def hay()
        return @donkey
    end

    def grass()
        return @elephant
    end

    def mouse()
        return @fox
    end
end


zoo = Zoo.new
s = 0
start = Time.now

while s < 10000000
    s = s + zoo.ant() + zoo.banana() + zoo.tuna() + zoo.hay() + zoo.grass() + zoo.mouse()
end

p(s)
p(Time.now - start)
